src/Client.php

```diff
@@ -55,7 +55,7 @@ public function __construct(MongoClientOptions $options, bool $useCoroutine = tr
      * Connect to Mongo using TCP/IP
      * and Wire Protocol.
      */
-    public function connect(): MongoClient
+    public function connect(): Client
```

Comment:
```suggestion
    public function connect(): self
```

Using `self` in all places allow us to avoid coupling with the class name.

Reply:
Should apply to all places.